core/src/components/datetime/datetime.tsx

```diff
@@ -844,85 +828,56 @@ export class Datetime implements ComponentInterface {
+          calendarBodyRef.scrollLeft = workingMonth.clientWidth * (isRTL(this.el) ? -1 : 1);
+          calendarBodyRef.style.removeProperty('overflow');
         });
       };
 
-      const threshold =
-        mode === 'ios' && typeof navigator !== 'undefined' && navigator.maxTouchPoints > 1 ? [0.7, 1] : 1;
-
-      // Intersection observers cannot accurately detect the
-      // intersection with a threshold of 1, when the observed
-      // element width is a sub-pixel value (i.e. 334.05px).
-      // Setting a root margin to 1px solves the issue.
-      const rootMargin = '1px';
+      /**
+       * When the container finishes scrolling we
+       * need to update the DOM with the selected month.
+       */
+      let scrollTimeout: ReturnType<typeof setTimeout> | undefined;
 
       /**
-       * Listen on the first month to
-       * prepend a new month and on the last
-       * month to append a new month.
-       * The 0.7 threshold is required on ios
-       * so that we can remove pointer-events
-       * when adding new months.
-       * Adding to a scroll snapping container
-       * while the container is snapping does not
-       * completely work as expected in WebKit.
-       * Adding pointer-events: none allows us to
-       * avoid these issues.
-       *
-       * This should be fine on Chromium, but
-       * when you set pointer-events: none
-       * it applies to active gestures which is not
-       * something WebKit does.
+       * We do not want to attempt to set pointer-events
+       * multiple times within a single swipe gesture as
+       * that adds unnecessary work to the main thread.
        */
+      let appliediOSRubberBandFix = false;
+      const scrollCallback = () => {
+        if (scrollTimeout) {
+          clearTimeout(scrollTimeout);
+        }
 
-      endIO = new IntersectionObserver((ev) => ioCallback('end', ev), {
-        threshold,
-        root: calendarBodyRef,
-        rootMargin,
-      });
-      endIO.observe(endMonth);
+        /**
+         * On iOS it is possible to quickly rubber band
+         * the scroll area before the scroll timeout has fired.
+         * This results in users reaching the end of the scrollable
+         * container before the DOM has updated.
+         * By setting `pointer-events: none` we can ensure that
+         * subsequent swipes do not happen while the container
+         * is snapping.
+         */
+        if (!appliediOSRubberBandFix && needsiOSRubberBandFix) {
+          calendarBodyRef.style.setProperty('pointer-events', 'none');
+          appliediOSRubberBandFix = true;
+        }
 
-      startIO = new IntersectionObserver((ev) => ioCallback('start', ev), {
-        threshold,
-        root: calendarBodyRef,
-        rootMargin,
-      });
-      startIO.observe(startMonth);
+        // Wait ~3 frames
+        scrollTimeout = setTimeout(updateActiveMonth, 50);
+      };
+      calendarBodyRef.addEventListener('scroll', scrollCallback);
```

Comment:
There was a similar issue in `main`, but we can probably fix this with a `touchstart/touchend` listener.

Reply:
Ok I fixed this. mind taking another look? I revised the `getChangedMonth` logic.

Instead of getting the month at the center point of `.calendar-body`, it gets the months at the left and right edges of the calendar body. If the month elements are the same, then we know that browser has finished snapping and the user has finished its gesture. If the months are not the same, it returns early and waits for the next scroll event.

It is possible for component to update when the user is still tapping the screen but has perfectly scrolled the month into view.

Reply:
Actually turns out `elementFromPoint` does not return values that are outside the viewport, so this causes the swipe logic to break if the page is zoom. I put up a new patch that should improve upon this logic.

The logic gets the element at the center of the calendar body. If the edges of the month and the calendar body do not line up (within a 2px tolerance), then the view does not update. This should account for users who are mid-swipe and have paused temporarily.

Reply:
Looks much better now, thank you! 🚀